cpp/src/arrow/dataset/scanner.h

```diff
@@ -198,10 +183,10 @@ class ARROW_DS_EXPORT Scanner {
 class ARROW_DS_EXPORT ScannerBuilder {
  public:
   ScannerBuilder(std::shared_ptr<Dataset> dataset,
-                 std::shared_ptr<ScanContext> scan_context);
+                 std::shared_ptr<ScanOptions> scan_options);
```

Comment:
Is it worthwhile to require ScannerBuilder be constructed with this `scan_options` argument? Maybe we could just have `explicit ScannerBuilder(std::shared_ptr<Dataset>)` construct a defaulted ScanOptions

Reply:
I've added a simpler constructor.